### DISCUSSION TITLE
Is there any proper way to upload files

### DISCUSSION THREAD

**Discussion by navan0** [fastapi/fastapi#7108] [Questions] [answered]
Is there any proper way to upload files and configure paths like in Flask ?

UPLOAD_FOLDER = '/path/to/the/uploads'
app.config['UPLOAD_FOLDER'] = UPLOAD_FOLDER

(https://flask.palletsprojects.com/en/1.1.x/patterns/fileuploads/)

**Comment by Mause** [comment]
https://fastapi.tiangolo.com/tutorial/request-files

**Comment by navan0** [comment]
Yeah, but for saving the files now I'm using ```shutil``` 

```
for idx, img in enumerate(images):
    with open(str(idx)+".jpg", "wb") as buffer:
        shutil.copyfileobj(img.file, buffer)
```

**Comment by ycd** [comment, marked as answer]
Because you are able to save the file `async` right now, Flask doesn't provide you an alternative. In FastAPI it is totally up to you, you can either use a coroutine or a function. Since you can use coroutine you can also use libraries like aiofiles.

```python
import aiofiles

@app.post("/uploadfile/")
async def cache_favicon(file: UploadFile = File(...)):
    async with aiofiles.open("destination.jpg" , "wb") as f:
        await f.write(await file.read())
```

**Comment by tiangolo** [comment]
Thanks for the help here everyone! :clap: :bow: 

Thanks for reporting back and closing the issue :+1:
